src/_pytest/cacheprovider.py

```diff
@@ -203,17 +203,32 @@ def set(self, key: str, value: object) -> None:
             with f:
                 f.write(data)
 
-    def _ensure_supporting_files(self) -> None:
-        """Create supporting files in the cache dir that are not really part of the cache."""
-        readme_path = self._cachedir / "README.md"
-        readme_path.write_text(README_CONTENT, encoding="UTF-8")
-
-        gitignore_path = self._cachedir.joinpath(".gitignore")
-        msg = "# Created by pytest automatically.\n*\n"
-        gitignore_path.write_text(msg, encoding="UTF-8")
+    def _ensure_cache_dir_and_supporting_files(self) -> None:
+        """Create the cache dir and its supporting files."""
+        if self._cachedir.is_dir():
+            return
 
-        cachedir_tag_path = self._cachedir.joinpath("CACHEDIR.TAG")
-        cachedir_tag_path.write_bytes(CACHEDIR_TAG_CONTENT)
+        with tempfile.TemporaryDirectory() as d:
```

Comment:
`d` is not a very descriptive name for the value, maybe `temp_dir` would be better?

Reply:
Before this change:
```
% git grep 'tempfile.TemporaryDirectory'
doc/en/how-to/fixtures.rst:        with tempfile.TemporaryDirectory() as newpath:
testing/test_assertrewrite.py:                    with tempfile.TemporaryDirectory() as d:
```
I was trying to be consistent. `temp_dir` would be wrong since the type is a string. What would you like?

Reply:
`newpath` looks good :)

Reply:
Done.